### PR TITLE
Fix fragmentation [7903]

### PR DESCRIPTION
**Pull request by julibert**
This PR fixes message fragmentation and alignment.
Before this PR, the `ucdrBuffer` given to users for publishing/subscribing had a non-zero offset which produces de/serialization error as was commented in https://github.com/eProsima/Micro-XRCE-DDS-Agent/issues/99.

Before merge:
- [x] merge https://github.com/eProsima/Micro-CDR/pull/51.
- [x] change `_microcdr_tag` to develop.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/97/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/101/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/103/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/107/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/104/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/108/)

File: include/uxr/client/core/session/session.h [hunk 1]
```diff
@@ -54,7 +55,7 @@ typedef void (*uxrOnRequestFunc) (
         uxrObjectId object_id,
         uint16_t request_id,
         SampleIdentity* sample_id,
-        uint8_t* buffer,
+        struct ucdrBuffer* ub,
         size_t len,
```
[the remark this answers is not in the document]
Comment (julibert) [reply include/uxr/client/core/session/session.h]: Because previously this callback had a `uint8_t` buffer.
I changed to `uint16_t`.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/105/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/109/)

**Pull request merged**